### ISSUE TITLE
[varnish::plugin] Varnish 4 not compatible

### ISSUE THREAD

**Issue by adrik976** [centreon/centreon-plugins#370]
Hello,
Varnish plugin is not working with Varnish 4 because of the output from varnishstat:

example:
before: backend_conn
now: MAIN.backend_conn

Is there any upgrade planned?
thanks

**Comment by garnier-quentin** [comment]
Could you give some examples of output (old one and new one) ?

Thanks.

**Comment by adrik976** [comment]
/usr/lib/nagios/plugins$ ./centreon_plugins.pl --plugin=apps::varnish::local::plugin --mode=uptime --hostname=varnish3 --remote --ssh-option='-l=user'
OK: Uptime in sec: 264489.80 | 'uptime'=264489.80;;;;

/usr/lib/nagios/plugins$ ./centreon_plugins.pl --plugin=apps::varnish::local::plugin --mode=uptime --hostname=varnish4 --remote --ssh-option='-l=user'
OK: Uptime in sec: 0.00 | 'uptime'=0.00;;;;

For varnish4 each value is at 0.

**Comment by garnier-quentin** [comment]
I mean the command used by the plugin (like the varnishstat ;)

**Comment by adrik976** [comment]
for varnish 3:
varnishstat -1
client_conn            2199214         0.08 Client connections accepted
client_drop                  0         0.00 Connection dropped, no sess/wrk
client_req              972192         0.04 Client requests received
cache_hit               115668         0.00 Cache hits
cache_hitpass                0         0.00 Cache hits for pass
cache_miss               31695         0.00 Cache misses
backend_conn             58714         0.00 Backend conn. success
backend_unhealthy            0         0.00 Backend conn. not attempted
backend_busy                 0         0.00 Backend conn. too many
backend_fail                 0         0.00 Backend conn. failures
backend_reuse             1941         0.00 Backend conn. reuses
backend_toolate          47358         0.00 Backend conn. was closed
backend_recycle          49300         0.00 Backend conn. recycles
backend_retry                3         0.00 Backend conn. retry
fetch_head                   0         0.00 Fetch head
fetch_length             60138         0.00 Fetch with Length
fetch_chunked                1         0.00 Fetch chunked
fetch_eof                    0         0.00 Fetch EOF
fetch_bad                    0         0.00 Fetch had bad headers
fetch_close                  7         0.00 Fetch wanted close
fetch_oldhttp                0         0.00 Fetch pre HTTP/1.1 closed
fetch_zero                   0         0.00 Fetch zero len
fetch_failed                 0         0.00 Fetch failed
fetch_1xx                    0         0.00 Fetch no body (1xx)
fetch_204                    0         0.00 Fetch no body (204)
fetch_304                    0         0.00 Fetch no body (304)
n_sess_mem                  14          .   N struct sess_mem
n_sess                       0          .   N struct sess
n_object                    17          .   N struct object
n_vampireobject              0          .   N unresurrected objects
n_objectcore                45          .   N struct objectcore
n_objecthead                33          .   N struct objecthead
n_waitinglist               32          .   N struct waitinglist
n_vbc                        1          .   N struct vbc
n_wrk                      800          .   N worker threads
n_wrk_create               800         0.00 N worker threads created
n_wrk_failed                 0         0.00 N worker threads not created
n_wrk_max                    0         0.00 N worker threads limited
n_wrk_lqueue                 0         0.00 work request queue length
n_wrk_queued                 0         0.00 N queued work requests
n_wrk_drop                   0         0.00 N dropped work requests
n_backend                    1          .   N backends
n_expired                31647          .   N expired objects
n_lru_nuked                  0          .   N LRU nuked objects
n_lru_moved             115565          .   N LRU moved objects
losthdr                      0         0.00 HTTP header overflows
n_objsendfile                0         0.00 Objects sent with sendfile
n_objwrite              971660         0.04 Objects sent with write
n_objoverflow                0         0.00 Objects overflowing workspace
s_sess                 2199214         0.08 Total Sessions
s_req                   972192         0.04 Total Requests
s_pipe                     431         0.00 Total pipe
s_pass                   28646         0.00 Total pass
s_fetch                  60146         0.00 Total fetch
s_hdrbytes           261004625        10.05 Total header bytes
s_bodybytes          807273837        31.08 Total body bytes
sess_closed            1242784         0.05 Session Closed
sess_pipeline                0         0.00 Session Pipeline
sess_readahead              14         0.00 Session Read Ahead
sess_linger             163600         0.01 Session Linger
sess_herd               957078         0.04 Session herd
shm_records           57795275         2.22 SHM records
shm_writes            23925809         0.92 SHM writes
shm_flushes                  0         0.00 SHM flushes due to overflow
shm_cont                   241         0.00 SHM MTX contention
shm_cycles                  22         0.00 SHM cycles through buffer
sms_nreq                796095         0.03 SMS allocator requests
sms_nobj                     0          .   SMS outstanding allocations
sms_nbytes                   0          .   SMS outstanding bytes
sms_balloc           796095705          .   SMS bytes allocated
sms_bfree            796095705          .   SMS bytes freed
backend_req              60161         0.00 Backend requests made
n_vcl                        2         0.00 N vcl total
n_vcl_avail                  2         0.00 N vcl available
n_vcl_discard                0         0.00 N vcl discarded
n_ban                        1          .   N total active bans
n_ban_gone                   0          .   N total gone bans
n_ban_add                   15         0.00 N new bans added
n_ban_retire                14         0.00 N old bans deleted
n_ban_obj_test             173         0.00 N objects tested
n_ban_re_test              184         0.00 N regexps tested against
n_ban_dups                   2         0.00 N duplicate bans removed
hcb_nolock              147363         0.01 HCB Lookups without lock
hcb_lock                  2655         0.00 HCB Lookups with lock
hcb_insert                2655         0.00 HCB Inserts
esi_errors                 146         0.00 ESI parse errors (unlock)
esi_warnings               146         0.00 ESI parse warnings (unlock)
accept_fail                  0         0.00 Accept failures
client_drop_late             0         0.00 Connection dropped late
uptime                25977773         1.00 Client uptime
dir_dns_lookups              0         0.00 DNS director lookups
dir_dns_failed               0         0.00 DNS director failed lookups
dir_dns_hit                  0         0.00 DNS director cached lookups hit
dir_dns_cache_full           0         0.00 DNS director full dnscache
vmods                        0          .   Loaded VMODs
n_gzip                   26055         0.00 Gzip operations
n_gunzip                206243         0.01 Gunzip operations
sess_pipe_overflow           0          .   Dropped sessions due to session pipe overflow
LCK.sms.creat                1         0.00 Created locks
LCK.sms.destroy              0         0.00 Destroyed locks
LCK.sms.locks          2388285         0.09 Lock Operations
LCK.sms.colls                0         0.00 Collisions
LCK.smp.creat                0         0.00 Created locks
LCK.smp.destroy              0         0.00 Destroyed locks
LCK.smp.locks                0         0.00 Lock Operations
LCK.smp.colls                0         0.00 Collisions
LCK.sma.creat                2         0.00 Created locks
LCK.sma.destroy              0         0.00 Destroyed locks
LCK.sma.locks          1893390         0.07 Lock Operations
LCK.sma.colls                0         0.00 Collisions
LCK.smf.creat                0         0.00 Created locks
LCK.smf.destroy              0         0.00 Destroyed locks
LCK.smf.locks                0         0.00 Lock Operations
LCK.smf.colls                0         0.00 Collisions
LCK.hsl.creat                0         0.00 Created locks
LCK.hsl.destroy              0         0.00 Destroyed locks
LCK.hsl.locks                0         0.00 Lock Operations
LCK.hsl.colls                0         0.00 Collisions
LCK.hcb.creat                1         0.00 Created locks
LCK.hcb.destroy              0         0.00 Destroyed locks
LCK.hcb.locks           149630         0.01 Lock Operations
LCK.hcb.colls                0         0.00 Collisions
LCK.hcl.creat                0         0.00 Created locks
LCK.hcl.destroy              0         0.00 Destroyed locks
LCK.hcl.locks                0         0.00 Lock Operations
LCK.hcl.colls                0         0.00 Collisions
LCK.vcl.creat                1         0.00 Created locks
LCK.vcl.destroy              0         0.00 Destroyed locks
LCK.vcl.locks               92         0.00 Lock Operations
LCK.vcl.colls                0         0.00 Collisions
LCK.stat.creat               1         0.00 Created locks
LCK.stat.destroy             0         0.00 Destroyed locks
LCK.stat.locks         2199228         0.08 Lock Operations
LCK.stat.colls               0         0.00 Collisions
LCK.sessmem.creat            1         0.00 Created locks
LCK.sessmem.destroy            0         0.00 Destroyed locks
LCK.sessmem.locks        2384691         0.09 Lock Operations
LCK.sessmem.colls              0         0.00 Collisions
LCK.wstat.creat                1         0.00 Created locks
LCK.wstat.destroy              0         0.00 Destroyed locks
LCK.wstat.locks         50100547         1.93 Lock Operations
LCK.wstat.colls                0         0.00 Collisions
LCK.herder.creat               1         0.00 Created locks
LCK.herder.destroy             0         0.00 Destroyed locks
LCK.herder.locks               1         0.00 Lock Operations
LCK.herder.colls               0         0.00 Collisions
LCK.wq.creat                  16         0.00 Created locks
LCK.wq.destroy                 0         0.00 Destroyed locks
LCK.wq.locks           419964066        16.17 Lock Operations
LCK.wq.colls                   0         0.00 Collisions
LCK.objhdr.creat            2687         0.00 Created locks
LCK.objhdr.destroy          2654         0.00 Destroyed locks
LCK.objhdr.locks       196527541         7.57 Lock Operations
LCK.objhdr.colls               0         0.00 Collisions
LCK.exp.creat                  1         0.00 Created locks
LCK.exp.destroy                0         0.00 Destroyed locks
LCK.exp.locks           26035932         1.00 Lock Operations
LCK.exp.colls                  0         0.00 Collisions
LCK.lru.creat                  2         0.00 Created locks
LCK.lru.destroy                0         0.00 Destroyed locks
LCK.lru.locks              31664         0.00 Lock Operations
LCK.lru.colls                  0         0.00 Collisions
LCK.cli.creat                  1         0.00 Created locks
LCK.cli.destroy                0         0.00 Destroyed locks
LCK.cli.locks            5006241         0.19 Lock Operations
LCK.cli.colls                  0         0.00 Collisions
LCK.ban.creat                  1         0.00 Created locks
LCK.ban.destroy                0         0.00 Destroyed locks
LCK.ban.locks          219892929         8.46 Lock Operations
LCK.ban.colls                  0         0.00 Collisions
LCK.vbp.creat                  1         0.00 Created locks
LCK.vbp.destroy                0         0.00 Destroyed locks
LCK.vbp.locks            5185797         0.20 Lock Operations
LCK.vbp.colls                  0         0.00 Collisions
LCK.vbe.creat                  1         0.00 Created locks
LCK.vbe.destroy                0         0.00 Destroyed locks
LCK.vbe.locks             117427         0.00 Lock Operations
LCK.vbe.colls                  0         0.00 Collisions
LCK.backend.creat              1         0.00 Created locks
LCK.backend.destroy            0         0.00 Destroyed locks
LCK.backend.locks         274740         0.01 Lock Operations
LCK.backend.colls              0         0.00 Collisions
SMA.s0.c_req              859479         0.03 Allocator requests
SMA.s0.c_fail                  0         0.00 Allocator failures
SMA.s0.c_bytes       31335588015      1206.25 Bytes allocated
SMA.s0.c_freed       31335570284      1206.25 Bytes freed
SMA.s0.g_alloc                34          .   Allocations outstanding
SMA.s0.g_bytes             17731          .   Bytes outstanding
SMA.s0.g_space        1073724093          .   Bytes available
SMA.Transient.c_req        57160         0.00 Allocator requests
SMA.Transient.c_fail           0         0.00 Allocator failures
SMA.Transient.c_bytes   3769515909       145.11 Bytes allocated
SMA.Transient.c_freed   3769515909       145.11 Bytes freed
SMA.Transient.g_alloc            0          .   Allocations outstanding
SMA.Transient.g_bytes            0          .   Bytes outstanding
SMA.Transient.g_space            0          .   Bytes available
VBE.machine(IP,,80).vcls            2          .   VCL references
VBE.machine(IP,,80).happy18446744073709551615          .   Happy health probes

for varnish 4:

varnishstat -1
MAIN.uptime            8435401         1.00 Child process uptime
MAIN.sess_conn          238398         0.03 Sessions accepted
MAIN.sess_drop               0         0.00 Sessions dropped
MAIN.sess_fail               0         0.00 Session accept failures
MAIN.sess_pipe_overflow            0         0.00 Session pipe overflow
MAIN.client_req_400                0         0.00 Client requests received, subject to 400 errors
MAIN.client_req_411                0         0.00 Client requests received, subject to 411 errors
MAIN.client_req_413                0         0.00 Client requests received, subject to 413 errors
MAIN.client_req_417                0         0.00 Client requests received, subject to 417 errors
MAIN.client_req               193371         0.02 Good client requests received
MAIN.cache_hit                 86448         0.01 Cache hits
MAIN.cache_hitpass                 0         0.00 Cache hits for pass
MAIN.cache_miss               106375         0.01 Cache misses
MAIN.backend_conn              61019         0.01 Backend conn. success
MAIN.backend_unhealthy             0         0.00 Backend conn. not attempted
MAIN.backend_busy                  0         0.00 Backend conn. too many
MAIN.backend_fail                  2         0.00 Backend conn. failures
MAIN.backend_reuse             45847         0.01 Backend conn. reuses
MAIN.backend_toolate           60878         0.01 Backend conn. was closed
MAIN.backend_recycle          106726         0.01 Backend conn. recycles
MAIN.backend_retry                 0         0.00 Backend conn. retry
MAIN.fetch_head                    0         0.00 Fetch no body (HEAD)
MAIN.fetch_length             105925         0.01 Fetch with Length
MAIN.fetch_chunked               937         0.00 Fetch chunked
MAIN.fetch_eof                     0         0.00 Fetch EOF
MAIN.fetch_bad                     0         0.00 Fetch bad T-E
MAIN.fetch_close                   0         0.00 Fetch wanted close
MAIN.fetch_oldhttp                 0         0.00 Fetch pre HTTP/1.1 closed
MAIN.fetch_zero                    0         0.00 Fetch zero len body
MAIN.fetch_1xx                     0         0.00 Fetch no body (1xx)
MAIN.fetch_204                     0         0.00 Fetch no body (204)
MAIN.fetch_304                     0         0.00 Fetch no body (304)
MAIN.fetch_failed                  0         0.00 Fetch failed (all causes)
MAIN.fetch_no_thread               0         0.00 Fetch failed (no thread)
MAIN.pools                         2          .   Number of thread pools
MAIN.threads                     400          .   Total number of threads
MAIN.threads_limited               0         0.00 Threads hit max
MAIN.threads_created             400         0.00 Threads created
MAIN.threads_destroyed             0         0.00 Threads destroyed
MAIN.threads_failed                0         0.00 Thread creation failed
MAIN.thread_queue_len              0          .   Length of session queue
MAIN.busy_sleep                    9         0.00 Number of requests sent to sleep on busy objhdr
MAIN.busy_wakeup                   9         0.00 Number of requests woken after sleep on busy objhdr
MAIN.sess_queued                   0         0.00 Sessions queued for thread
MAIN.sess_dropped                  0         0.00 Sessions dropped for thread
MAIN.n_object                     40          .   object structs made
MAIN.n_vampireobject               0          .   unresurrected objects
MAIN.n_objectcore                 48          .   objectcore structs made
MAIN.n_objecthead                 51          .   objecthead structs made
MAIN.n_waitinglist                17          .   waitinglist structs made
MAIN.n_backend                     1          .   Number of backends
MAIN.n_expired                106335          .   Number of expired objects
MAIN.n_lru_nuked                   0          .   Number of LRU nuked objects
MAIN.n_lru_moved               85969          .   Number of LRU moved objects
MAIN.losthdr                       0         0.00 HTTP header overflows
MAIN.s_sess                   238398         0.03 Total sessions seen
MAIN.s_req                    193371         0.02 Total requests seen
MAIN.s_pipe                        0         0.00 Total pipe sessions seen
MAIN.s_pass                      548         0.00 Total pass-ed requests seen
MAIN.s_fetch                  106923         0.01 Total backend fetches initiated
MAIN.s_synth                       0         0.00 Total synthethic responses made
MAIN.s_req_hdrbytes         25311455         3.00 Request header bytes
MAIN.s_req_bodybytes          560108         0.07 Request body bytes
MAIN.s_resp_hdrbytes        49477897         5.87 Response header bytes
MAIN.s_resp_bodybytes      485523647        57.56 Response body bytes
MAIN.s_pipe_hdrbytes               0         0.00 Pipe request header bytes
MAIN.s_pipe_in                     0         0.00 Piped bytes from client
MAIN.s_pipe_out                    0         0.00 Piped bytes to client
MAIN.sess_closed              141947         0.02 Session Closed
MAIN.sess_pipeline                 0         0.00 Session Pipeline
MAIN.sess_readahead                0         0.00 Session Read Ahead
MAIN.sess_herd                  3037         0.00 Session herd
MAIN.shm_records            24309202         2.88 SHM records
MAIN.shm_writes              8012449         0.95 SHM writes
MAIN.shm_flushes                   0         0.00 SHM flushes due to overflow
MAIN.shm_cont                   9638         0.00 SHM MTX contention
MAIN.shm_cycles                    9         0.00 SHM cycles through buffer
MAIN.sms_nreq                      0         0.00 SMS allocator requests
MAIN.sms_nobj                      0          .   SMS outstanding allocations
MAIN.sms_nbytes                    0          .   SMS outstanding bytes
MAIN.sms_balloc                    0          .   SMS bytes allocated
MAIN.sms_bfree                     0          .   SMS bytes freed
MAIN.backend_req              106862         0.01 Backend requests made
MAIN.n_vcl                         1         0.00 Number of loaded VCLs in total
MAIN.n_vcl_avail                   1         0.00 Number of VCLs available
MAIN.n_vcl_discard                 0         0.00 Number of discarded VCLs
MAIN.bans                          1          .   Count of bans
MAIN.bans_completed                1          .   Number of bans marked 'completed'
MAIN.bans_obj                      0          .   Number of bans using obj.*
MAIN.bans_req                      0          .   Number of bans using req.*
MAIN.bans_added                    1         0.00 Bans added
MAIN.bans_deleted                  0         0.00 Bans deleted
MAIN.bans_tested                   0         0.00 Bans tested against objects (lookup)
MAIN.bans_obj_killed               0         0.00 Objects killed by bans (lookup)
MAIN.bans_lurker_tested            0         0.00 Bans tested against objects (lurker)
MAIN.bans_tests_tested             0         0.00 Ban tests tested against objects (lookup)
MAIN.bans_lurker_tests_tested            0         0.00 Ban tests tested against objects (lurker)
MAIN.bans_lurker_obj_killed              0         0.00 Objects killed by bans (lurker)
MAIN.bans_dups                           0         0.00 Bans superseded by other bans
MAIN.bans_lurker_contention              0         0.00 Lurker gave way for lookup
MAIN.bans_persisted_bytes               13          .   Bytes used by the persisted ban lists
MAIN.bans_persisted_fragmentation            0          .   Extra bytes in persisted ban lists due to fragmentation
MAIN.n_purges                                0          .   Number of purge operations executed
MAIN.n_obj_purged                            0          .   Number of purged objects
MAIN.exp_mailed                         106375         0.01 Number of objects mailed to expiry thread
MAIN.exp_received                       106375         0.01 Number of objects received by expiry thread
MAIN.hcb_nolock                         192823         0.02 HCB Lookups without lock
MAIN.hcb_lock                           106374         0.01 HCB Lookups with lock
MAIN.hcb_insert                         106374         0.01 HCB Inserts
MAIN.esi_errors                              0         0.00 ESI parse errors (unlock)
MAIN.esi_warnings                            0         0.00 ESI parse warnings (unlock)
MAIN.vmods                                   2          .   Loaded VMODs
MAIN.n_gzip                             106507         0.01 Gzip operations
MAIN.n_gunzip                           188549         0.02 Gunzip operations
MAIN.vsm_free                           972544          .   Free VSM space
MAIN.vsm_used                         83962064          .   Used VSM space
MAIN.vsm_cooling                             0          .   Cooling VSM space
MAIN.vsm_overflow                            0          .   Overflow VSM space
MAIN.vsm_overflowed                          0         0.00 Overflowed VSM space
MGT.uptime                             8435401         1.00 Management process uptime
MGT.child_start                              1         0.00 Child process started
MGT.child_exit                               0         0.00 Child process normal exit
MGT.child_stop                               0         0.00 Child process unexpected exit
MGT.child_died                               0         0.00 Child process died (signal)
MGT.child_dump                               0         0.00 Child process core dumped
MGT.child_panic                              0         0.00 Child process panic
MEMPOOL.vbc.live                             1          .   In use
MEMPOOL.vbc.pool                            10          .   In Pool
MEMPOOL.vbc.sz_wanted                       88          .   Size requested
MEMPOOL.vbc.sz_needed                      120          .   Size allocated
MEMPOOL.vbc.allocs                       61021         0.01 Allocations
MEMPOOL.vbc.frees                        61020         0.01 Frees
MEMPOOL.vbc.recycle                      61021         0.01 Recycled from pool
MEMPOOL.vbc.timeout                        199         0.00 Timed out from pool
MEMPOOL.vbc.toosmall                         0         0.00 Too small to recycle
MEMPOOL.vbc.surplus                          0         0.00 Too many for pool
MEMPOOL.vbc.randry                           0         0.00 Pool ran dry
MEMPOOL.busyobj.live                         0          .   In use
MEMPOOL.busyobj.pool                        10          .   In Pool
MEMPOOL.busyobj.sz_wanted                65536          .   Size requested
MEMPOOL.busyobj.sz_needed                65568          .   Size allocated
MEMPOOL.busyobj.allocs                  106923         0.01 Allocations
MEMPOOL.busyobj.frees                   106923         0.01 Frees
MEMPOOL.busyobj.recycle                 106923         0.01 Recycled from pool
MEMPOOL.busyobj.timeout                    763         0.00 Timed out from pool
MEMPOOL.busyobj.toosmall                     0         0.00 Too small to recycle
MEMPOOL.busyobj.surplus                      0         0.00 Too many for pool
MEMPOOL.busyobj.randry                       0         0.00 Pool ran dry
MEMPOOL.req0.live                            0          .   In use
MEMPOOL.req0.pool                           10          .   In Pool
MEMPOOL.req0.sz_wanted                   65536          .   Size requested
MEMPOOL.req0.sz_needed                   65568          .   Size allocated
MEMPOOL.req0.allocs                     119835         0.01 Allocations
MEMPOOL.req0.frees                      119835         0.01 Frees
MEMPOOL.req0.recycle                    119835         0.01 Recycled from pool
MEMPOOL.req0.timeout                       616         0.00 Timed out from pool
MEMPOOL.req0.toosmall                        0         0.00 Too small to recycle
MEMPOOL.req0.surplus                         0         0.00 Too many for pool
MEMPOOL.req0.randry                          0         0.00 Pool ran dry
MEMPOOL.sess0.live                           0          .   In use
MEMPOOL.sess0.pool                          10          .   In Pool
MEMPOOL.sess0.sz_wanted                    384          .   Size requested
MEMPOOL.sess0.sz_needed                    416          .   Size allocated
MEMPOOL.sess0.allocs                    119184         0.01 Allocations
MEMPOOL.sess0.frees                     119184         0.01 Frees
MEMPOOL.sess0.recycle                   119184         0.01 Recycled from pool
MEMPOOL.sess0.timeout                     1173         0.00 Timed out from pool
MEMPOOL.sess0.toosmall                       0         0.00 Too small to recycle
MEMPOOL.sess0.surplus                        0         0.00 Too many for pool
MEMPOOL.sess0.randry                         0         0.00 Pool ran dry
MEMPOOL.req1.live                            0          .   In use
MEMPOOL.req1.pool                           10          .   In Pool
MEMPOOL.req1.sz_wanted                   65536          .   Size requested
MEMPOOL.req1.sz_needed                   65568          .   Size allocated
MEMPOOL.req1.allocs                     119933         0.01 Allocations
MEMPOOL.req1.frees                      119933         0.01 Frees
MEMPOOL.req1.recycle                    119933         0.01 Recycled from pool
MEMPOOL.req1.timeout                       627         0.00 Timed out from pool
MEMPOOL.req1.toosmall                        0         0.00 Too small to recycle
MEMPOOL.req1.surplus                         0         0.00 Too many for pool
MEMPOOL.req1.randry                          0         0.00 Pool ran dry
MEMPOOL.sess1.live                           0          .   In use
MEMPOOL.sess1.pool                          10          .   In Pool
MEMPOOL.sess1.sz_wanted                    384          .   Size requested
MEMPOOL.sess1.sz_needed                    416          .   Size allocated
MEMPOOL.sess1.allocs                    119214         0.01 Allocations
MEMPOOL.sess1.frees                     119214         0.01 Frees
MEMPOOL.sess1.recycle                   119214         0.01 Recycled from pool
MEMPOOL.sess1.timeout                     1201         0.00 Timed out from pool
MEMPOOL.sess1.toosmall                       0         0.00 Too small to recycle
MEMPOOL.sess1.surplus                        0         0.00 Too many for pool
MEMPOOL.sess1.randry                         0         0.00 Pool ran dry
SMA.default.c_req                       212788         0.03 Allocator requests
SMA.default.c_fail                           0         0.00 Allocator failures
SMA.default.c_bytes                  864055406       102.43 Bytes allocated
SMA.default.c_freed                  841093835        99.71 Bytes freed
SMA.default.g_alloc                         80          .   Allocations outstanding
SMA.default.g_bytes                   22961571          .   Bytes outstanding
SMA.default.g_space                  245473885          .   Bytes available
SMA.Transient.c_req                       1153         0.00 Allocator requests
SMA.Transient.c_fail                         0         0.00 Allocator failures
SMA.Transient.c_bytes                  6840474         0.81 Bytes allocated
SMA.Transient.c_freed                  6840474         0.81 Bytes freed
SMA.Transient.g_alloc                        0          .   Allocations outstanding
SMA.Transient.g_bytes                        0          .   Bytes outstanding
SMA.Transient.g_space                        0          .   Bytes available
VBE.machine(IP,,82).vcls            1          .   VCL references
VBE.machine(IP,,82).happy 18446744073709551615          .   Happy health probes
VBE.machine(IP,,82).bereq_hdrbytes     26784649         3.18 Request header bytes
VBE.machine(IP,,82).bereq_bodybytes       561146         0.07 Request body bytes
VBE.machine(IP,,82).beresp_hdrbytes     17887388         2.12 Response header bytes
VBE.machine(IP,,82).beresp_bodybytes    410317978        48.64 Response body bytes
VBE.machine(IP,,82).pipe_hdrbytes               0         0.00 Pipe request header bytes
VBE.machine(IP,,82).pipe_out                    0         0.00 Piped bytes to backend
VBE.machine(IP,,82).pipe_in                     0         0.00 Piped bytes from backend
LCK.sms.creat                                                  0         0.00 Created locks
LCK.sms.destroy                                                0         0.00 Destroyed locks
LCK.sms.locks                                                  0         0.00 Lock Operations
LCK.smp.creat                                                  0         0.00 Created locks
LCK.smp.destroy                                                0         0.00 Destroyed locks
LCK.smp.locks                                                  0         0.00 Lock Operations
LCK.sma.creat                                                  2         0.00 Created locks
LCK.sma.destroy                                                0         0.00 Destroyed locks
LCK.sma.locks                                             427802         0.05 Lock Operations
LCK.smf.creat                                                  0         0.00 Created locks
LCK.smf.destroy                                                0         0.00 Destroyed locks
LCK.smf.locks                                                  0         0.00 Lock Operations
LCK.hsl.creat                                                  0         0.00 Created locks
LCK.hsl.destroy                                                0         0.00 Destroyed locks
LCK.hsl.locks                                                  0         0.00 Lock Operations
LCK.hcb.creat                                                  1         0.00 Created locks
LCK.hcb.destroy                                                0         0.00 Destroyed locks
LCK.hcb.locks                                             259572         0.03 Lock Operations
LCK.hcl.creat                                                  0         0.00 Created locks
LCK.hcl.destroy                                                0         0.00 Destroyed locks
LCK.hcl.locks                                                  0         0.00 Lock Operations
LCK.vcl.creat                                                  1         0.00 Created locks
LCK.vcl.destroy                                                0         0.00 Destroyed locks
LCK.vcl.locks                                             422637         0.05 Lock Operations
LCK.sessmem.creat                                              0         0.00 Created locks
LCK.sessmem.destroy                                            0         0.00 Destroyed locks
LCK.sessmem.locks                                              0         0.00 Lock Operations
LCK.sess.creat                                            238398         0.03 Created locks
LCK.sess.destroy                                          238398         0.03 Destroyed locks
LCK.sess.locks                                               546         0.00 Lock Operations
LCK.wstat.creat                                                1         0.00 Created locks
LCK.wstat.destroy                                              0         0.00 Destroyed locks
LCK.wstat.locks                                           787631         0.09 Lock Operations
LCK.herder.creat                                               0         0.00 Created locks
LCK.herder.destroy                                             0         0.00 Destroyed locks
LCK.herder.locks                                               0         0.00 Lock Operations
LCK.wq.creat                                                   3         0.00 Created locks
LCK.wq.destroy                                                 0         0.00 Destroyed locks
LCK.wq.locks                                             4413061         0.52 Lock Operations
LCK.objhdr.creat                                          106383         0.01 Created locks
LCK.objhdr.destroy                                        106331         0.01 Destroyed locks
LCK.objhdr.locks                                         1845476         0.22 Lock Operations
LCK.exp.creat                                                  1         0.00 Created locks
LCK.exp.destroy                                                0         0.00 Destroyed locks
LCK.exp.locks                                             819466         0.10 Lock Operations
LCK.lru.creat                                                  2         0.00 Created locks
LCK.lru.destroy                                                0         0.00 Destroyed locks
LCK.lru.locks                                             409784         0.05 Lock Operations
LCK.cli.creat                                                  1         0.00 Created locks
LCK.cli.destroy                                                0         0.00 Destroyed locks
LCK.cli.locks                                            2811267         0.33 Lock Operations
LCK.ban.creat                                                  1         0.00 Created locks
LCK.ban.destroy                                                0         0.00 Destroyed locks
LCK.ban.locks                                           27908049         3.31 Lock Operations
LCK.vbp.creat                                                  1         0.00 Created locks
LCK.vbp.destroy                                                0         0.00 Destroyed locks
LCK.vbp.locks                                             562339         0.07 Lock Operations
LCK.backend.creat                                              1         0.00 Created locks
LCK.backend.destroy                                            0         0.00 Destroyed locks
LCK.backend.locks                                         396513         0.05 Lock Operations
LCK.vcapace.creat                                              1         0.00 Created locks
LCK.vcapace.destroy                                            0         0.00 Destroyed locks
LCK.vcapace.locks                                              0         0.00 Lock Operations
LCK.nbusyobj.creat                                             0         0.00 Created locks
LCK.nbusyobj.destroy                                           0         0.00 Destroyed locks
LCK.nbusyobj.locks                                             0         0.00 Lock Operations
LCK.busyobj.creat                                         106923         0.01 Created locks
LCK.busyobj.destroy                                       106923         0.01 Destroyed locks
LCK.busyobj.locks                                         692327         0.08 Lock Operations
LCK.mempool.creat                                              6         0.00 Created locks
LCK.mempool.destroy                                            0         0.00 Destroyed locks
LCK.mempool.locks                                       46127476         5.47 Lock Operations
LCK.vxid.creat                                                 1         0.00 Created locks
LCK.vxid.destroy                                               0         0.00 Destroyed locks
LCK.vxid.locks                                                27         0.00 Lock Operations
LCK.pipestat.creat                                             1         0.00 Created locks
LCK.pipestat.destroy                                           0         0.00 Destroyed locks
LCK.pipestat.locks                                             0         0.00 Lock Operations

**Comment by florian-asche** [comment]
Hi @adrik976 if you help find the differences (missing), i can make a pull request and fix this. 
@garnier-quentin i can add the missing in $maps_counters. It is then duplicate in $maps_counters but if the entry is only once in data, its only once handled.

**Comment by adrik976** [comment]
I think the problem is that the script is searching for an exact match for example: backend_conn but in the varnistat command it's not the same output, we have MAIN.backend_conn.
I try something out to check differences:

foreach (keys %{$maps_counters}) {
        print $self->{result}->{"MAIN".'.'.$_}."\n";
        print $self->{result}->{$_}."\n";
        };

**Comment by florian-asche** [comment]
Do you have an update?

**Comment by Sims24** [comment]
@florian-asche  could you provide to Quentin or I output in my last message here : 
https://github.com/centreon/centreon-plugins/issues/534

**Comment by thatsk** [comment]
what should be warning and critical parameter would be set 
for this.

**Comment by garnier-quentin** [comment]
I can't provide the thresholds. It depends of your varnish usage.

**Comment by thatsk** [comment]
{code}
 perl /usr/lib64/nagios/plugins/mobi/centreon_plugins.pl --plugin=apps::varnish::local::plugin --mode=backend
OK: Backend con: 0.00/s, Backend unhealthy: 0.00/s, Backend busy: 0.00/s, Backend fail: 0.00/s, Backend reuse: 0.00/s, Backend recycle: 0.00/s, Backend retry: 0.00/s, Backend recycle: 0.00/s, Backend requests: 0.00/s | 'backend_conn'=0.00;;;0; 'backend_unhealthy'=0.00;;;0; 'backend_busy'=0.00;;;0; 'backend_fail'=0.00;;;0; 'backend_reuse'=0.00;;;0; 'backend_recycle'=0.00;;;0; 'backend_retry'=0.00;;;0; 'backend_recycle'=0.00;;;0; 'backend_req'=0.00;;;0;
{code}

**Comment by thatsk** [comment]
@garnier-quentin 
session mode is not outputting the output.
perl /usr/lib64/nagios/plugins/mobi/centreon_plugins.pl --plugin=apps::varnish::local::plugin --mode=sessions
OK:  |

```BASH
varnishstat -1  2>&1
client_conn              15879         0.02 Client connections accepted
client_drop                  0         0.00 Connection dropped, no sess/wrk
client_req               15879         0.02 Client requests received
cache_hit                    0         0.00 Cache hits
cache_hitpass                0         0.00 Cache hits for pass
cache_miss                   0         0.00 Cache misses
backend_conn                 0         0.00 Backend conn. success
backend_unhealthy            0         0.00 Backend conn. not attempted
backend_busy                 0         0.00 Backend conn. too many
backend_fail                 0         0.00 Backend conn. failures
backend_reuse                0         0.00 Backend conn. reuses
backend_toolate              0         0.00 Backend conn. was closed
backend_recycle              0         0.00 Backend conn. recycles
backend_retry                0         0.00 Backend conn. retry
fetch_head                   0         0.00 Fetch head
fetch_length                 0         0.00 Fetch with Length
fetch_chunked                0         0.00 Fetch chunked
fetch_eof                    0         0.00 Fetch EOF
fetch_bad                    0         0.00 Fetch had bad headers
fetch_close                  0         0.00 Fetch wanted close
fetch_oldhttp                0         0.00 Fetch pre HTTP/1.1 closed
fetch_zero                   0         0.00 Fetch zero len
fetch_failed                 0         0.00 Fetch failed
fetch_1xx                    0         0.00 Fetch no body (1xx)
fetch_204                    0         0.00 Fetch no body (204)
fetch_304                    0         0.00 Fetch no body (304)
n_sess_mem                  10          .   N struct sess_mem
n_sess                       0          .   N struct sess
n_object                     0          .   N struct object
n_vampireobject              0          .   N unresurrected objects
n_objectcore                 4          .   N struct objectcore
n_objecthead                 4          .   N struct objecthead
n_waitinglist                4          .   N struct waitinglist
n_vbc                        0          .   N struct vbc
n_wrk                      400          .   N worker threads
n_wrk_create               400         0.00 N worker threads created
n_wrk_failed                 0         0.00 N worker threads not created
n_wrk_max                    0         0.00 N worker threads limited
n_wrk_lqueue                 0         0.00 work request queue length
n_wrk_queued                 0         0.00 N queued work requests
n_wrk_drop                   0         0.00 N dropped work requests
n_backend                   19          .   N backends
n_expired                    0          .   N expired objects
n_lru_nuked                  0          .   N LRU nuked objects
n_lru_moved                  0          .   N LRU moved objects
losthdr                      0         0.00 HTTP header overflows
n_objsendfile                0         0.00 Objects sent with sendfile
n_objwrite               15879         0.02 Objects sent with write
n_objoverflow                0         0.00 Objects overflowing workspace
s_sess                   15879         0.02 Total Sessions
s_req                    15879         0.02 Total Requests
s_pipe                       0         0.00 Total pipe
s_pass                       0         0.00 Total pass
s_fetch                      0         0.00 Total fetch
s_hdrbytes             3334590         3.50 Total header bytes
s_bodybytes             444612         0.47 Total body bytes
sess_closed              15879         0.02 Session Closed
sess_pipeline                0         0.00 Session Pipeline
sess_readahead               0         0.00 Session Read Ahead
sess_linger                  0         0.00 Session Linger
sess_herd                    0         0.00 Session herd
shm_records            2302966         2.42 SHM records
shm_writes             1810717         1.90 SHM writes
shm_flushes                  0         0.00 SHM flushes due to overflow
shm_cont                  1115         0.00 SHM MTX contention
shm_cycles                   2         0.00 SHM cycles through buffer
sms_nreq                 15879         0.02 SMS allocator requests
sms_nobj                     0          .   SMS outstanding allocations
sms_nbytes                   0          .   SMS outstanding bytes
sms_balloc              444612          .   SMS bytes allocated
sms_bfree               444612          .   SMS bytes freed
backend_req                  0         0.00 Backend requests made
n_vcl                       20         0.00 N vcl total
n_vcl_avail                 20         0.00 N vcl available
n_vcl_discard                0         0.00 N vcl discarded
n_ban                        1          .   N total active bans
n_ban_gone                   1          .   N total gone bans
n_ban_add                    1         0.00 N new bans added
n_ban_retire                 0         0.00 N old bans deleted
n_ban_obj_test               0         0.00 N objects tested
n_ban_re_test                0         0.00 N regexps tested against
n_ban_dups                   0         0.00 N duplicate bans removed
hcb_nolock                   0         0.00 HCB Lookups without lock
hcb_lock                     0         0.00 HCB Lookups with lock
hcb_insert                   0         0.00 HCB Inserts
esi_errors                   0         0.00 ESI parse errors (unlock)
esi_warnings                 0         0.00 ESI parse warnings (unlock)
accept_fail                  0         0.00 Accept failures
client_drop_late             0         0.00 Connection dropped late
uptime                  952627         1.00 Client uptime
dir_dns_lookups              0         0.00 DNS director lookups
dir_dns_failed               0         0.00 DNS director failed lookups
dir_dns_hit                  0         0.00 DNS director cached lookups hit
dir_dns_cache_full           0         0.00 DNS director full dnscache
vmods                        0          .   Loaded VMODs
n_gzip                       0         0.00 Gzip operations
n_gunzip                     0         0.00 Gunzip operations
sess_pipe_overflow           0          .   Dropped sessions due to session pipe overflow
LCK.sms.creat                1         0.00 Created locks
LCK.sms.destroy              0         0.00 Destroyed locks
LCK.sms.locks            47637         0.05 Lock Operations
LCK.sms.colls                0         0.00 Collisions
LCK.smp.creat                0         0.00 Created locks
LCK.smp.destroy              0         0.00 Destroyed locks
LCK.smp.locks                0         0.00 Lock Operations
LCK.smp.colls                0         0.00 Collisions
LCK.sma.creat                2         0.00 Created locks
LCK.sma.destroy              0         0.00 Destroyed locks
LCK.sma.locks            31758         0.03 Lock Operations
LCK.sma.colls                0         0.00 Collisions
LCK.smf.creat                0         0.00 Created locks
LCK.smf.destroy              0         0.00 Destroyed locks
LCK.smf.locks                0         0.00 Lock Operations
LCK.smf.colls                0         0.00 Collisions
LCK.hsl.creat                0         0.00 Created locks
LCK.hsl.destroy              0         0.00 Destroyed locks
LCK.hsl.locks                0         0.00 Lock Operations
LCK.hsl.colls                0         0.00 Collisions
LCK.hcb.creat                1         0.00 Created locks
LCK.hcb.destroy              0         0.00 Destroyed locks
LCK.hcb.locks             5293         0.01 Lock Operations
LCK.hcb.colls                0         0.00 Collisions
LCK.hcl.creat                0         0.00 Created locks
LCK.hcl.destroy              0         0.00 Destroyed locks
LCK.hcl.locks                0         0.00 Lock Operations
LCK.hcl.colls                0         0.00 Collisions
LCK.vcl.creat                1         0.00 Created locks
LCK.vcl.destroy              0         0.00 Destroyed locks
LCK.vcl.locks              148         0.00 Lock Operations
LCK.vcl.colls                0         0.00 Collisions
LCK.stat.creat               1         0.00 Created locks
LCK.stat.destroy             0         0.00 Destroyed locks
LCK.stat.locks           15889         0.02 Lock Operations
LCK.stat.colls               0         0.00 Collisions
LCK.sessmem.creat            1         0.00 Created locks
LCK.sessmem.destroy            0         0.00 Destroyed locks
LCK.sessmem.locks          17873         0.02 Lock Operations
LCK.sessmem.colls              0         0.00 Collisions
LCK.wstat.creat                1         0.00 Created locks
LCK.wstat.destroy              0         0.00 Destroyed locks
LCK.wstat.locks          1909798         2.00 Lock Operations
LCK.wstat.colls                0         0.00 Collisions
LCK.herder.creat               1         0.00 Created locks
LCK.herder.destroy             0         0.00 Destroyed locks
LCK.herder.locks               1         0.00 Lock Operations
LCK.herder.colls               0         0.00 Collisions
LCK.wq.creat                   4         0.00 Created locks
LCK.wq.destroy                 0         0.00 Destroyed locks
LCK.wq.locks             3841162         4.03 Lock Operations
LCK.wq.colls                   0         0.00 Collisions
LCK.objhdr.creat               4         0.00 Created locks
LCK.objhdr.destroy             0         0.00 Destroyed locks
LCK.objhdr.locks               0         0.00 Lock Operations
LCK.objhdr.colls               0         0.00 Collisions
LCK.exp.creat                  1         0.00 Created locks
LCK.exp.destroy                0         0.00 Destroyed locks
LCK.exp.locks             952252         1.00 Lock Operations
LCK.exp.colls                  0         0.00 Collisions
LCK.lru.creat                  2         0.00 Created locks
LCK.lru.destroy                0         0.00 Destroyed locks
LCK.lru.locks                  0         0.00 Lock Operations
LCK.lru.colls                  0         0.00 Collisions
LCK.cli.creat                  1         0.00 Created locks
LCK.cli.destroy                0         0.00 Destroyed locks
LCK.cli.locks              16900         0.02 Lock Operations
LCK.cli.colls                  0         0.00 Collisions
LCK.ban.creat                  1         0.00 Created locks
LCK.ban.destroy                0         0.00 Destroyed locks
LCK.ban.locks             952253         1.00 Lock Operations
LCK.ban.colls                  0         0.00 Collisions
LCK.vbp.creat                  1         0.00 Created locks
LCK.vbp.destroy                0         0.00 Destroyed locks
LCK.vbp.locks            1713371         1.80 Lock Operations
LCK.vbp.colls                  0         0.00 Collisions
LCK.vbe.creat                  1         0.00 Created locks
LCK.vbe.destroy                0         0.00 Destroyed locks
LCK.vbe.locks                  0         0.00 Lock Operations
LCK.vbe.colls                  0         0.00 Collisions
LCK.backend.creat             19         0.00 Created locks
LCK.backend.destroy            0         0.00 Destroyed locks
LCK.backend.locks              0         0.00 Lock Operations
LCK.backend.colls              0         0.00 Collisions
SMA.s0.c_req               15879         0.02 Allocator requests
SMA.s0.c_fail                  0         0.00 Allocator failures
SMA.s0.c_bytes         541664448       568.60 Bytes allocated
SMA.s0.c_freed         541664448       568.60 Bytes freed
SMA.s0.g_alloc                 0          .   Allocations outstanding
SMA.s0.g_bytes                 0          .   Bytes outstanding
SMA.s0.g_space        8589934592          .   Bytes available
SMA.Transient.c_req            0         0.00 Allocator requests
SMA.Transient.c_fail           0         0.00 Allocator failures
SMA.Transient.c_bytes            0         0.00 Bytes allocated
SMA.Transient.c_freed            0         0.00 Bytes freed
SMA.Transient.g_alloc            0          .   Allocations outstanding
SMA.Transient.g_bytes            0          .   Bytes outstanding
SMA.Transient.g_space            0          .   Bytes available
```

**Comment by thatsk** [comment]
@garnier-quentin how to set for client connections or something which is understandble so that i can set threshold by understanding output

**Comment by thatsk** [comment]
@garnier-quentin 
this one too  so i am bit confused about setting up threshold.
perl /usr/lib64/nagios/plugins/mobi/centreon_plugins.pl --plugin=apps::varnish::local::plugin --mode=uptime --warning 0 --critical 50
CRITICAL: Uptime in sec: 63530.80 | 'uptime'=63530.80;0:0;0:50;;

**Comment by garnier-quentin** [comment]
There is no session mode. Please use the last version. (mode=stats)

**Comment by thatsk** [comment]
@garnier-quentin can you guide for threshold

**Comment by thatsk** [comment]
there is not mode available.
```BASH
perl /usr/lib64/nagios/plugins/mobi/centreon_plugins.pl --plugin=apps::varnish::local::plugin --list-mode

Plugin Description:
    Check Varnish Cache with varnishstat Command

Global Options:
    --mode  Choose a mode.

    --dyn-mode
            Specify a mode with the path (separated by '::').

    --list-mode
            List available modes.

    --mode-version
            Check minimal version of mode. If not, unknown error.

    --version
            Display plugin version.

Modes Available:
   backend
   bans
   cache
   clients
   connections
   dns
   esi
   fetch
   hcb
   n
   objects
   sessions
   shm
   sms
   threads
   totals
   uptime
   vcl
   workers
```

**Comment by garnier-quentin** [comment]
It's a old version. Please use the new one.

**Comment by thatsk** [comment]
@garnier-quentin can you just help me out through threshold on one of mode.
